tests/app/shared/comment-editor/comment-editor.component.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { CUSTOM_ELEMENTS_SCHEMA, DebugElement } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormControl, FormGroup, FormsModule } from '@angular/forms';
+import { By } from '@angular/platform-browser';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { Apollo } from 'apollo-angular';
+import { MarkdownModule } from 'ngx-markdown';
+
+import { CommentEditorComponent } from '../../../../src/app/shared/comment-editor/comment-editor.component';
+import { SharedModule } from '../../../../src/app/shared/shared.module';
+
+describe('CommentEditor', () => {
+  let fixture: ComponentFixture<CommentEditorComponent>;
+  let debugElement: DebugElement;
+  let component: CommentEditorComponent;
+
+  const TEST_INITIAL_DESCRIPTION = 'abc';
+  const TEST_SUBMIT_BUTTON_TEXT = 'Submit';
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [CommentEditorComponent],
+      imports: [FormsModule, SharedModule, MarkdownModule.forRoot(), BrowserAnimationsModule],
+      providers: [Apollo],
+      schemas: [CUSTOM_ELEMENTS_SCHEMA]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(CommentEditorComponent);
+    debugElement = fixture.debugElement;
+    component = fixture.componentInstance;
+
+    // initialize compulsory inputs
+    const commentField: FormControl = new FormControl('');
+    const commentForm: FormGroup = new FormGroup({
+      description: commentField
+    });
+    const id = 'description';
+
+    // manually inject inputs into the component
+    component.commentField = commentField;
+    component.commentForm = commentForm;
+    component.id = id;
+    component.submitButtonText = TEST_SUBMIT_BUTTON_TEXT;
+  });
+
+  describe('text box is rendered correctly', () => {
```

Comment:
Just a nitpick regarding syntax, according to https://jasmine.github.io/tutorials/your_first_suite, for describe function, ```The string parameter is for naming the collection of specs, and will be concatenated with specs to make a spec's full name. This aids in finding specs in a large suite. If you name them well, your specs read as full sentences in traditional [BDD](http://en.wikipedia.org/wiki/Behavior-driven_development) style.```

Reply:
You can refer to https://github.com/CATcher-org/CATcher/pull/977 for reference!

Reply:
Thanks for pointing this out @kkangs0226 😃

Just made some changes to the naming of this `describe` and some `it` statements for better readability.